blenderset/assets.py

```diff
@@ -81,6 +85,29 @@ def get_all_objects_of_class(self, class_name):
                 objs.append(obj)
         return objs
 
+    def get_roi(self):
+        if self.override_roi is not None:
+            return self.override_roi
+        walkable_polys = self.get_all_proprty_values("blenderset.visible_walkable_roi")
+        return MultiPolygon([Polygon(p) for p in walkable_polys])
+
+    def random_position(self, roi):
+        if roi.bounds and all(np.isfinite(roi.bounds)):
+            min_x, min_y, max_x, max_y = roi.bounds
+            while True:
```

Comment:
Possible infinite loop if the polygon is bad. Please add break condition or change to a for loop.